src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,191 @@
+            throw new DukeException("OOPS!!! Please include the time of event after /at.");
+        }
+    }
+
+    /**
+     * This method checks to ensure that the user provides
+     * a keyword to search for from the task list.
+     *
+     * @param tasks The user's input split into an array of strings.
+     * @throws DukeException If there is no keyword to search for.
+     */
+    private static void findCheck(String[] tasks) throws DukeException {
+        if (tasks.length <= 1) {
+            throw new DukeException("OOPS!!! Please tell me what you want to find!!");
+        }
+    }
+
+    /**
+     * This method when called reformats the date input by the user to a Date object.
+     *
+     * @param date The date string to be reformatted.
+     * @return The <code>Date</code> in a formatted manner.
+     * @throws DukeException If there is a problem with data processing, loading or saving.
+     */
+    public static Date dateFormatter(String date) throws DukeException {
+        try {
+            SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
+            Date parseDate = formatter.parse(date);
+            return parseDate;
+        } catch (ParseException e) {
+            throw new DukeException("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    /**
+     * This method when called will result in the returning of a <code>Command</code>
+     * which can be executed to result in either a deletion of the task, an addition of
+     * a task or to mark a task as done from the task list.
+     *
+     * @param command The user's input string.
+     * @return A class of command depending on the type of user input.
+     * @throws DukeException If there is a problem with data processing, loading or saving.
+     */
+    public static Command parse(String command) throws DukeException {
+        if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        } else {
+            String[] task = command.split(" ");
+            String instruction = task[0];
+
+            if (instruction.equals("done")) {
+                int taskNumber = Integer.parseInt(task[1]);
+                return new DoneCommand(taskNumber - 1);
+            } else if (instruction.equals("delete")) {
+                try {
+                    int taskNumber = Integer.parseInt(task[1]);
+                    return new DeleteCommand(taskNumber - 1);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new DukeException("OOPS!!! Provide a task number to delete!");
+                }
+            } else {
+                switch (instruction) {
+                case "todo": {
+                    try {
+                        todoCheck(task);
+                        Task todo = new Todo(command.substring(5));
```

Comment:
What happens if the user decides to enter "todo                                        sleep" instead of "todo sleep"?

Reply:
Good point but I do not actually agree with this, for if a user were to do that, he might be keen to format it like that on purpose. If I want to include 8 spaces just because I am keen to do so, but the programme keeps restricting me from doing so, then it is a bit too annoying. Most to-do lists out there do not have such a behaviour.

Reply:
That's actually a good point that you raised. At first, I was thinking that it would be better to check whether commands like "todo __________ sleep" would be valid. But to think about it, it might actually be possible for us to accept such commands while ensuring that the output syntax aligns with how it should be like. One possible way which I thought of was:

String toDoName =command.substring(5).trim();  // trim() helps to remove any leading and trailing whitespaces.
Task todo = new Todo(toDoName);

Sounds like a good thing for us to implement. Instead of outright rejecting the user's input, we can help them to trim it as long as they input the correct details in the correct order.